### ISSUE TITLE
curl: security update to 8.1.0

### ISSUE THREAD

**Issue by MingcongBai** [AOSC-Dev/aosc-os-abbs#4549]
### CVE IDs

- CVE-2022-43551, CVE-2022-43552
- CVE-2023-23914, CVE-2023-23915, CVE-2023-23916
- CVE-2023-27533, CVE-2023-27534, CVE-2023-27535, CVE-2023-27536, CVE-2023-27538
- CVE-2023-28319, CVE-2023-28320, CVE-2023-28321, CVE-2023-28322

### Other security advisory IDs

N/A

### Description

Several security vulnerabilities were discovered & fixed between curl 7.86.0 and 8.1.0. Details are listed below:

- CVE-2023-28322: more POST-after-PUT confusion 
- CVE-2023-28321: IDN wildcard match
- CVE-2023-28320: siglongjmp race conditio
- CVE-2023-28319: UAF in SSH sha256 fingerprint check
- CVE-2023-27538: SSH connection too eager reuse still 
- CVE-2023-27537: HSTS double free 
- CVE-2023-27536: GSS delegation too eager connection re-use 
- CVE-2023-27535: FTP too eager connection reuse 
- CVE-2023-27534: SFTP path ~ resolving discrepancy 
- CVE-2023-27533: TELNET option IAC injecti
- CVE-2023-23916: HTTP multi-header compression denial of - service 
- CVE-2023-23915: HSTS amnesia with --parallel 
- CVE-2023-23914: HSTS ignored on multiple requests 
- CVE-2022-43552: HTTP Proxy deny use after free 
- CVE-2022-43551: Another HSTS bypass via IDN 

### Patches

In short, an update to version 8.1.0 will fix these vulnerabilities. 

### PoC(s)

N/A
